cmd/ci-secret-bootstrap/main.go

```diff
@@ -621,27 +621,35 @@ func mutateGlobalPullSecret(original, secret *coreapi.Secret) (bool, error) {
 	if err != nil {
 		return false, fmt.Errorf("failed to parse the constructed secret: %w", err)
 	}
-	registryDomain := api.DomainForService(api.ServiceRegistry)
-	if dockerConfig.Auths == nil || dockerConfig.Auths[api.DomainForService(api.ServiceRegistry)].Auth == "" {
-		return false, fmt.Errorf("failed to get token for %s", registryDomain)
+	if len(dockerConfig.Auths) == 0 {
+		return false, fmt.Errorf("failed to get any token")
 	}
-	token := dockerConfig.Auths[api.DomainForService(api.ServiceRegistry)].Auth
-	dockerConfig, err = dockerConfigJSON(original)
+	originalDockerConfig, err := dockerConfigJSON(original)
 	if err != nil {
 		return false, fmt.Errorf("failed to parse the original secret: %w", err)
 	}
-	if dockerConfig.Auths[api.DomainForService(api.ServiceRegistry)].Auth == token {
-		return false, nil
-	}
-	dockerConfig.Auths[api.DomainForService(api.ServiceRegistry)] = secretbootstrap.DockerAuth{
-		Auth: token,
+	mutatedSecret := false
+	domains := []string{api.DomainForService(api.ServiceRegistry), api.QCIAPPCIDomain, api.QuayOpenShiftCIRepo, api.QCICache}
+	for _, domain := range domains {
+		if dockerConfig.Auths[domain].Auth == "" {
+			return false, fmt.Errorf("failed to get token for %s", domain)
+		}
+		originalToken := originalDockerConfig.Auths[domain].Auth
+		originalDockerConfig.Auths[domain] = secretbootstrap.DockerAuth{
+			Auth: dockerConfig.Auths[domain].Auth,
+		}
+		if originalToken != originalDockerConfig.Auths[domain].Auth {
+			mutatedSecret = true
+		}
 	}
-	data, err := json.Marshal(dockerConfig)
-	if err != nil {
-		return false, fmt.Errorf("failed to marshal the docker config: %w", err)
+	if mutatedSecret {
+		data, err := json.Marshal(dockerConfig)
```

Comment:
```suggestion
		data, err := json.Marshal(originalDockerConfig)
```

This is an important one i guess.
The key here is that we would like to keep the change minimal, i.e., we modify the secret ONLY when it is necessary and we modify only the parts that belong to us.